aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -515,6 +515,368 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
   return result;
 }
 
+// helper methods for matrix_exp
+namespace {
+
+template <typename scalar_t, int ROW, int COL>
+using array2d = std::array<std::array<scalar_t, COL>, ROW>;
+
+// we consider 6 Taylor expansions of degree
+// 1, 2, 4, 8, 12, 18
+constexpr int total_n_degs = 6;
+
+// 0! to 4!
+// Introduced to avoid 'magic number's clang-tidy complaints about
+constexpr int fact_array_size = 5;
+constexpr std::array<float, fact_array_size> fact = {1., 1., 2., 6., 24.};
+
+// Iterates over first `n_batch_dims` and applies `f` to each element.
+// TODO: this implementation is not efficient, it simply iterates over
+// batch elements in a serial for-loop.
```

Comment:
I am sure we have many, many copies of this functionality all over the codebase XD

Reply:
This applies f in place, I guess?

Reply:
Sorry, I will update the documentation. It iterates over a list of tensors and applies `f` to the selected list of tensors. The batched functionality I saw only dealt with a single tensor.

Reply:
Yes, `f` is in-place.

Reply:
It does the essentially the same loop as TensorIterator.